skimage/segmentation/_watershed.py

```diff
@@ -85,32 +85,36 @@ def _validate_inputs(image, markers, mask, connectivity):
 
 def watershed(image, markers=None, connectivity=1, offset=None, mask=None,
               compactness=0, watershed_line=False):
-    """Find watershed basins in `image` flooded from given `markers`.
+    """Find watershed basins in an image flooded from given markers.
 
     Parameters
     ----------
     image : ndarray (2-D, 3-D, ...)
         Data array where the lowest value points are labeled first.
-    markers : int, or ndarray of int, same shape as `image`, optional
-        The desired number of markers, or an array marking the basins with the
+    markers : int, or ndarray of int with same shape as image, optional
+        The desired number of basins, or an array marking the basins with the
         values to be assigned in the label matrix. Zero means not a marker. If
-        ``None`` (no markers given), the local minima of the image are used as
-        markers.
+        None (no markers given), the local minima of the image are used as
+        markers. Note that, in this case, the local minima are determined in
+        such a way that it is equivalent to applying
+        :func:`skimage.morphology.local_minima` onto ``image`` followed by
+        :func:`skimage.morphology.label` onto the result
+        only if setting ``connectivity=2``.
```

Comment:
I personally find this really difficult to comprehend. Perhaps, it might be better to simplify this description as much as possible and encourage users to provide their own `markers` to improve reproducibility?
Otherwise, the PR looks great!

Reply:
That reads better! I think there is a possibility to go even further:
```
None, the (default) markers are determined as the local minima
 of ``image``. Specifically, the computation is equivalent
 to applying :func:`skimage.morphology.local_minima` onto 
``image``, followed by :func:`skimage.morphology.label` (with 
``connectivity=2``) onto the result. Generally, users are 
encouraged to pass markers explicitly. 
```
Please, let me know what you think.

Reply:
@soupault I think it reads well, thanks. Now, I'm wondering:

* I forget whether we want to encourage using `skimage.morphology.label` or `skimage.measure.label` (or maybe there's no preference?);
* I guess ``connectivity=2`` should be replaced with ``connectivity=image.ndim``?

Reply:
>`skimage.morphology.label` or `skimage.measure.label`

They refer to the same function in code, which is under `skimage.measure.`, so I would point to that.

>I guess connectivity=2 should be replaced with connectivity=image.ndim?

I have checked the code in details - the correct way seems to be to just say that the functions are called with the specified `connectivity`.

In the original issue, the local minima are calculated via `locMin = label(local_minima(im))`, with both functions using default `connectivity=im.ndim`, thus the confusion.

Reply:
Thanks for checking, @soupault! :pray: 

> They refer to the same function in code, which is under `skimage.measure.`, so I would point to that.

:+1: :ok_hand: 

> I have checked the code in details - the correct way seems to be to just say that the functions are called with the specified `connectivity` [...] default `connectivity=im.ndim`, thus the confusion.

Oh, right, right.